lib/Dialect/LWE/IR/LWETypes.td

```diff
@@ -46,7 +46,17 @@ def RLWECiphertext : LWE_Type<"RLWECiphertext", "rlwe_ciphertext"> {
   );
 }
 
-def RLWECiphertextLike : TypeOrContainer<RLWECiphertext, "ciphertext-like">;
+def RNSRLWECiphertext : LWE_Type<"RNSRLWECiphertext", "rns_rlwe_ciphertext"> {
+  let summary = "A type for RLWE ciphertexts";
+  let parameters = (ins
+    "::mlir::Attribute":$encoding,
+    "RNSRLWEParamsAttr":$rlwe_params,
```

Comment:
I think a good approach to avoiding duplication here would be to have `RLWEParams` take a more generic attribute for the ring. Instead of a ring attr, it could be a generic Attribute with a verifier, or [`AnyAttributeOf` constraint](https://github.com/llvm/llvm-project/blob/458fa12202868281f11680a18878d57defcccc9f/mlir/include/mlir/IR/CommonAttrConstraints.td#L171) for the legal RNS/non-RNS ring attirbute options.

Reply:
Referencing #876 as I guess this should be taken into consideration in the LWE Attr&Types overhaul :)